src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/SetAzureSqlDatabase.cs

```diff
@@ -146,26 +182,65 @@ protected override IEnumerable<AzureSqlDatabaseModel> GetEntity()
         protected override IEnumerable<AzureSqlDatabaseModel> ApplyUserInputToModel(IEnumerable<AzureSqlDatabaseModel> model)
         {
             List<Model.AzureSqlDatabaseModel> newEntity = new List<AzureSqlDatabaseModel>();
+
             if (this.ParameterSetName == UpdateParameterSetName)
             {
-                newEntity.Add(new AzureSqlDatabaseModel()
+                AzureSqlDatabaseModel newDbModel = new AzureSqlDatabaseModel()
                 {
                     ResourceGroupName = ResourceGroupName,
                     ServerName = ServerName,
                     DatabaseName = DatabaseName,
-                    Edition = Edition,
                     MaxSizeBytes = MaxSizeBytes,
-                    RequestedServiceObjectiveName = RequestedServiceObjectiveName,
                     Tags = TagsConversionHelper.ReadOrFetchTags(this, model.FirstOrDefault().Tags),
                     ElasticPoolName = ElasticPoolName,
                     Location = model.FirstOrDefault().Location,
                     ReadScale = ReadScale,
                     ZoneRedundant =
                         MyInvocation.BoundParameters.ContainsKey("ZoneRedundant")
-                            ? (bool?) ZoneRedundant.ToBool()
+                            ? (bool?)ZoneRedundant.ToBool()
                             : null
-                });
+                };
+
+                if (!string.IsNullOrWhiteSpace(RequestedServiceObjectiveName) || MyInvocation.BoundParameters.ContainsKey("Edition"))
+                {
+                    newDbModel.Sku = new Management.Sql.Models.Sku()
+                    {
+                        Name = string.IsNullOrWhiteSpace(RequestedServiceObjectiveName) ? Edition.ToString() : RequestedServiceObjectiveName,
```

Comment:
@payiAzure make sure to do a null-check on `Edition` before using it here

Reply:
@cormacpayne The outer check "if (!string.IsNullOrWhiteSpace(RequestedServiceObjectiveName) || MyInvocation.BoundParameters.ContainsKey("Edition"))"
makes sure that two parameters RequestedServiceObjectiveName and Edition, one of them must not be null. So when use them here in line 208, if one is null, the other must have value I think.